modin/core/dataframe/pandas/dataframe/dataframe.py

```diff
@@ -983,6 +982,9 @@ def from_labels(self) -> "PandasDataframe":
                 if "index" not in self.columns
                 else "level_{}".format(0)
             ]
+        names = tuple(level_names) if len(level_names) > 1 else level_names[0]
+        new_dtypes = self.index.to_frame(name=names).dtypes
+        new_dtypes = pandas.concat([new_dtypes, self.dtypes])
```

Comment:
I'm wondering should we do it all the time or only when the `self._dtypes` cache is accessible?

Reply:
Good point! I think we should compute `dtypes` only when `_dtypes` is accessible.